src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -760,7 +770,14 @@ bool GCToOSInterface::VirtualDecommit(void* address, size_t size)
     // that much more clear to the operating system that we no
     // longer need these pages. Also, GC depends on re-committed pages to
     // be zeroed-out.
-    return mmap(address, size, PROT_NONE, MAP_FIXED | MAP_ANON | MAP_PRIVATE, -1, 0) != NULL;
+    void* bRetVal = mmap(address, size, PROT_NONE, MAP_FIXED | MAP_ANON | MAP_PRIVATE, -1, 0);
```

Comment:
The decommit logic here looks incomplete when compared to PAL `VirtualFree` here: https://github.com/dotnet/runtime/blob/10d342cecc5cbfc57419f8a92af4787f29d76c08/src/coreclr/pal/src/map/virtual.cpp#L1556

Reply:
Yes, the madvise call with MADV_DONTDUMP is needed here.

Reply:
yeah have added that, but wonder if logic under `MMAP_ANON_IGNORES_PROTECTION` is required too. Also the retval should be checking for `MAP_FAILED` instead of `NULL` here.

Reply:
You are right regarding the checking of MAP_FAILED instead of NULL. As for the MMAP_ANON_IGNORES_PROTECTION, I don't think there is any platform where this would be detected as needed by the cmake configuration.

Reply:
made the changes now.